### PR TITLE
add support to rhel-87

### PR DESCRIPTION
**Pull request by acosferreira**
# Description

add support to rhel-87

Fixes # (THEEDGE-2807)

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: config/repo_config.go [hunk 1]
```diff
@@ -26,6 +26,7 @@ var DistributionsPackages = map[string][]string{
 	"rhel-84": RHEL8,
 	"rhel-85": RHEL8,
 	"rhel-86": RHEL86,
+	"rhel-87": RHEL86,
```
Comment (omaciel) [inline config/repo_config.go:29]: -"rhel-87": RHEL86,
+"rhel-87": RHEL87,

**Review by omaciel — CHANGES_REQUESTED** [review]
/lgtm

File: config/repo_config.go (hunk 1 above)
Comment (omaciel) [reply config/repo_config.go:29]: Anything that can make the code easier to maintain gets a 👍🏻  from me

File: config/repo_config.go [hunk 2]
```diff
@@ -12,20 +12,21 @@ var RequiredPackages = []string{"rhc",
...
-// RHEL86 contains aditional list of packages to build an image to = RHEL86
-var RHEL86 = []string{"ansible-core"}
+// RHEL8X contains additional list of packages to build an image to = RHEL86
+var RHEL8X = []string{"ansible-core"}
 
-// RHEL90 contains aditional list of packages to build an image to = RHEL90
+// RHEL90 contains additional list of packages to build an image to = RHEL90
 var RHEL90 = []string{"ansible-core"}
 
 // DistributionsPackages add packages byi mage
```
Comment (omaciel) [inline config/repo_config.go:24]: Minor typo: s/byi/by

File: config/repo_config_test.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+	distri9 := []string{"rhel-90"}
+
+	Context("Validate distribution ref values", func() {
+		It("should return RHEL8 distribution refs", func() {
+			for _, d := range distri8 {
+				Expect(config.DistributionsRefs[d]).ToNot(BeNil())
+				Expect(config.DistributionsRefs[d]).To(ContainSubstring("rhel"))
+				Expect(config.DistributionsRefs[d]).To(ContainSubstring("x86_64"))
+				Expect(config.DistributionsRefs[d]).To(ContainSubstring("edge"))
+				Expect(config.DistributionsRefs[d]).To(ContainSubstring("8"))
```
Comment (ldjebran) [inline config/repo_config_test.go:24]: I would prefer to be more explicit eg:

rhel-84", "rhel-85, "rhel-86", "rhel-87"  all have same ref , so one test is sufficient 

```go
Expect(config.DistributionsRefs[d]).To(Equal("rhel/8/x86_64/edge"))
```

what do you think ?

File: config/repo_config.go [hunk 4]
```diff
@@ -12,20 +12,21 @@ var RequiredPackages = []string{"rhc",
 	"subscription-manager-plugin-ostree",
 	"insights-client"}
 
-// RHEL8 contains aditional list of packages to build an image to >= RHEL85
+// RHEL8 contains additional list of packages to build an image to >= RHEL85
 var RHEL8 = []string{"ansible"}
 
-// RHEL86 contains aditional list of packages to build an image to = RHEL86
-var RHEL86 = []string{"ansible-core"}
+// RHEL8X contains additional list of packages to build an image to = RHEL86
```
Comment (loadtheaccumulator) [inline config/repo_config.go:18]: Named RHEL8X but the comment refers to RHEL86.

File: config/repo_config.go [hunk 5]
```diff
@@ -12,20 +12,21 @@ var RequiredPackages = []string{"rhc",
...
+var RHEL8X = []string{"ansible-core"}
 
-// RHEL90 contains aditional list of packages to build an image to = RHEL90
+// RHEL90 contains additional list of packages to build an image to = RHEL90
 var RHEL90 = []string{"ansible-core"}
 
-// DistributionsPackages add packages byi mage
+// DistributionsPackages add packages by mage
```
Comment (loadtheaccumulator) [inline config/repo_config.go:24]: mage typo

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]
just a couple of minor things

File: config/repo_config_test.go (hunk 3 above)
Comment (acosferreira) [reply config/repo_config_test.go:24]: make sense ;) 
thanks

File: config/repo_config_test.go [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+import (
+	"strings"
+	"testing"
+
+	"github.com/redhatinsights/edge-api/config"
+)
+
+// Validate distribution ref values should return refs by distribution"
+func TestValidateRepo(t *testing.T) {
+	supportedDistributions := []string{"rhel-84", "rhel-85", "rhel-86", "rhel-87", "rhel-90"}
```
Comment (omaciel) [inline config/repo_config_test.go:14]: Use the values from `DistributionsPackages` instead

File: config/repo_config_test.go [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+		if !strings.Contains(config.DistributionsRefs[d], majorVersion) {
+			t.Errorf("%q not found: %q", majorVersion, config.DistributionsRefs[d])
+		}
+
+	}
+
+}
+func TestValidatePackages(t *testing.T) {
+	supportedDistributions := []string{"rhel-84", "rhel-85", "rhel-86", "rhel-87", "rhel-90"}
```
Comment (omaciel) [inline config/repo_config_test.go:34]: Use the values from `DistributionsPackages` instead

File: config/repo_config_test.go [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	for _, d := range supportedDistributions {
+
+		if len(config.DistributionsPackages[d]) == 0 {
+			t.Errorf("package not found: %q", config.DistributionsPackages[d])
+		}
+	}
+
+}
+func TestSupportedPackages(t *testing.T) {
+	supportedDistributions := []string{"rhel-84", "rhel-85", "rhel-86", "rhel-87", "rhel-90"}
```
Comment (omaciel) [inline config/repo_config_test.go:45]: Use the values from `DistributionsPackages` instead

**Review by omaciel — CHANGES_REQUESTED** [review]
How about a test for each distribution's ref?

**Review by omaciel — DISMISSED** [review]
/lgtm

File: config/repo_config_test.go [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+func TestSupportedPackages(t *testing.T) {
+	t.Run("validate size", func(t *testing.T) {
+		if len(config.DistributionsPackages) != len(config.DistributionsRefs) {
+			t.Errorf("packages supported %d and found %d", len(config.DistributionsPackages), len(config.DistributionsRefs))
+		}
+	})
+
+}
+
+func TestInvaliddistribution(t *testing.T) {
```
Comment (loadtheaccumulator) [inline config/repo_config_test.go:55]: This should probably be TestInvalidDistribution.

File: config/repo_config_test.go [hunk 10]
```diff
@@ -0,0 +1,62 @@
+// FIXME: golangci-lint
+// nolint:errcheck,gosec,govet,revive,typecheck
+package config_test
```
Comment (loadtheaccumulator) [inline config/repo_config_test.go:3]: config_test instead of config?

File: config/config_suite_test.go [hunk 11]
```diff
@@ -0,0 +1,15 @@
+// FIXME: golangci-lint
+// nolint:revive,typecheck
+package config_test
```
Comment (loadtheaccumulator) [inline config/config_suite_test.go:3]: config_test instead of config package?

**Review by loadtheaccumulator — CHANGES_REQUESTED** [review]

**Review by omaciel — APPROVED** [review]
/lgtm

**Review by thom-at-redhat — APPROVED** [review]

**Review by AaronH88 — APPROVED** [review]
/lgtm

**Review by de1987 — APPROVED** [review]

**Pull request merged**